src/_png.cpp

```diff
@@ -206,7 +206,17 @@ static PyObject *Py_write_png(PyObject *self, PyObject *args, PyObject *kwds)
             goto exit;
         }
         buff.cursor = 0;
-    } else if ((fp = mpl_PyFile_Dup(py_file, (char *)"wb", &offset))) {
+    } else {
+        #if PY3K
```

Comment:
Nope -- this won't break.  Here's the definition of it in mpl_utils.h:

```c++
#if PY_MAJOR_VERSION >= 3
#define PY3K 1
#define Py_TPFLAGS_HAVE_NEWBUFFER 0
#else
#define PY3K 0
#endif
```

I take `PY3K` to mean "next-gen Python" as it was called before Python 3.0 was even released.